source/decryptor/decryptor.c

```diff
@@ -348,6 +434,64 @@ u32 GetNandCtr(u8* ctr, u32 offset)
     return 0;
 }
 
+u32 SeekFileInNand(u32* offset, u32* size, u32* seekpos, const char* filename, PartitionInfo* partition)
+{
+    // poor mans NAND FAT file seeker:
+    // - can't handle long filenames
+    // - filename must be in FAT 8+3 format
+    // - doesn't search the root dir
+    // - dirs must not exceed 1024 entries
+    // - fragmentation not supported
+    
+    const static char* magic = ".          ";
+    const static char zeroes[8+3] = { 0x00 };
+    u8* buffer = BUFFER_ADDRESS;
+    u32 p_size = partition->size;
+    u32 p_offset = partition->offset;
+    
+    u32 cluster_size;
+    u32 cluster_start;
+    u32 found = 0;
+    
+    
+    if (strlen(filename) != 8+3)
+        return 1;
+    
+    DecryptNandToMem(buffer, p_offset, NAND_SECTOR_SIZE, partition);
+    cluster_start =
+        NAND_SECTOR_SIZE * ( *((u16*) (buffer + 0x0E)) + // FAT table start
+        ( *((u16*) (buffer + 0x16)) * buffer[0x10] ) ) + // FAT table size
+        *((u16*) (buffer + 0x11)) * 0x20; // root directory size
+    cluster_size = buffer[0x0D] * NAND_SECTOR_SIZE;
+    if (seekpos != NULL) {
+        if (cluster_start > *seekpos)
+            *seekpos = cluster_start;
+    }
+    
+    for( u32 i = (seekpos == NULL) ? cluster_start : *seekpos; i < p_size; i += cluster_size ) {
+        DecryptNandToMem(buffer, p_offset + i, NAND_SECTOR_SIZE, partition);
+        if (memcmp(buffer, magic, 8+3) == 0) {
```

Comment:
Can you reduce the level of indentations in this function by `continue`ing instead of going down the `if` chain?